presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloMetadata.java

```diff
@@ -0,0 +1,427 @@
+            client.createOrReplaceView(viewName, viewData);
+        }
+        else {
+            client.createView(viewName, viewData);
+        }
+    }
+
+    @Override
+    public void dropView(ConnectorSession session, SchemaTableName viewName)
+    {
+        client.dropView(viewName);
+    }
+
+    @Override
+    public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> bldr = ImmutableMap.builder();
+        for (SchemaTableName stName : listViews(session, prefix.getSchemaName())) {
+            bldr.put(stName, new ConnectorViewDefinition(stName, Optional.empty(), client.getView(stName).getData()));
+        }
+        return bldr.build();
+    }
+
+    @Override
+    public List<SchemaTableName> listViews(ConnectorSession session, String schemaNameOrNull)
+    {
+        return listViews(schemaNameOrNull);
+    }
+
+    /**
+     * Gets all views in the given schema, or all schemas if null
+     *
+     * @param schemaNameOrNull Schema to list for the views, or null to list all schemas
+     * @return List of views
+     */
+    private List<SchemaTableName> listViews(String schemaNameOrNull)
+    {
+        ImmutableList.Builder<SchemaTableName> bldr = ImmutableList.builder();
+
+        if (schemaNameOrNull == null) {
+            for (String schema : client.getSchemaNames()) {
+                for (String view : client.getViewNames(schema)) {
+                    bldr.add(new SchemaTableName(schema, view));
+                }
+            }
+        }
+        else {
+            for (String view : client.getViewNames(schemaNameOrNull)) {
+                bldr.add(new SchemaTableName(schemaNameOrNull, view));
+            }
+        }
+
+        return bldr.build();
+    }
+
+    @Override
+    public ConnectorInsertTableHandle beginInsert(ConnectorSession session,
+            ConnectorTableHandle tableHandle)
+    {
+        checkNoRollback();
+        AccumuloTableHandle handle = checkType(tableHandle, AccumuloTableHandle.class, "table");
+        setRollback(() -> rollbackInsert(handle));
+        return handle;
+    }
+
+    @Override
+    public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle, Collection<Slice> fragments)
+    {
+        clearRollback();
+    }
+
+    private void rollbackInsert(ConnectorInsertTableHandle insertHandle)
+    {
+        // Rollbacks for inserts are off the table when it comes to data in Accumulo.
+        // When a batch of Mutations fails to be inserted, the general strategy
+        // is to run the insert operation again until it is successful
+        // Any mutations that were successfully written will be overwritten
+        // with the same values, so that isn't a problem.
+        throw new PrestoException(NOT_SUPPORTED, "Unable to rollback insert. Some rows may have been written. Please run your insert again.");
```

Comment:
Include table name in error message in case of more complex queries / longer transactions without auto-commit.